tests/integrationv2/test_renegotiate.py

```diff
@@ -258,4 +260,86 @@ def test_s2n_client_renegotiate_with_openssl(managed_process, cipher, curve, cer
         for output in Msg.expected_output(RENEG_MESSAGES, Provider.ClientMode):
             assert output in results.stdout
         assert renegotiate_was_successful(results)
+
+
+"""
+Renegotiation request with client auth accepted by s2n-tls client.
+
+The openssl server does not require client auth during the first handshake,
+but does require client auth during the second handshake.
+"""
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+@pytest.mark.parametrize("cipher", ALL_TEST_CIPHERS, ids=get_parameter_name)
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES, ids=get_parameter_name)
+@pytest.mark.parametrize("certificate", MINIMAL_TEST_CERTS, ids=get_parameter_name)
+@pytest.mark.parametrize("protocol", TEST_PROTCOLS, ids=get_parameter_name)
+@pytest.mark.parametrize("provider", [OpenSSL], ids=get_parameter_name)
+def test_s2n_client_renegotiate_with_client_auth_with_openssl(managed_process, cipher, curve, certificate, protocol, provider):
+    messages = RENEG_MESSAGES.copy()
+    ctrl_msg = [ m for m in messages if m.ctrl ][0]
+    messages[messages.index(ctrl_msg)] = Msg(Provider.ServerMode, ctrl_str='R\n')
```

Comment:
Can't you do something like:
`ctrl_msg.ctrl_str = 'R\n'`
I get what you have works it just looks so weird 😢

Reply:
I can't do that exactly because even after I copy RENEG_MESSAGES, it contains the same Msgs as the original. So changing the ctrl message changes the original ctrl message too. Think of them as array of pointers, not arrays of data.

But it looks like `copy.deepcopy()` can just make a copy of everything, so then I can modify it instead of recreate it!

Reply:
And I guess I could have just deep copied it myself with `messages = [ m.copy() for m in messages ]`. I'm not a python person either lol.